tools/downloader/downloader.py

```diff
@@ -43,53 +43,46 @@ def process_download(chunk_iterable, size, file):
                 duration = time.monotonic() - start_time
                 progress_size += len(chunk)
                 if duration != 0:
-                    speed = progress_size // (1024 * duration)
+                    speed = int(progress_size / (1024 * duration))
                     percent = str(progress_size * 100 // size)
 
-                    print('... %s%%, %d KB, %d KB/s, %d seconds passed' %
-                            (percent, progress_size / 1024, speed, duration),
-                        end='\r' if sys.stdout.isatty() else '\n', flush=True)
+                    reporter.print_progress('... {}%, {} KB, {} KB/s, {} seconds passed',
+                        percent, int(progress_size / 1024), speed, int(duration))
+                    reporter.emit_event('model_file_download_progress', size=progress_size)
 
                 file.write(chunk)
     finally:
-        if sys.stdout.isatty():
-            print()
+        reporter.end_progress()
 
-def try_download(file, num_attempts, start_download, size):
+def try_download(reporter, file, num_attempts, start_download, size):
     for attempt in range(num_attempts):
         if attempt != 0:
             retry_delay = 10
-            print("Will retry in {} seconds...".format(retry_delay))
+            reporter.print("Will retry in {} seconds...", retry_delay, flush=True)
             time.sleep(retry_delay)
 
         try:
             chunk_iterable = start_download()
             file.seek(0)
             file.truncate()
-            process_download(chunk_iterable, size, file)
+            process_download(reporter, chunk_iterable, size, file)
             return True
-        except requests.exceptions.ConnectionError as e:
-            print("Error Connecting:", e)
-        except requests.exceptions.Timeout as e:
-            print("Timeout Error:", e)
-        except requests.exceptions.TooManyRedirects as e:
-            print("Redirects Error: requests exceeds maximum number of redirects", e)
-        except (requests.exceptions.RequestException, ssl.SSLError) as e:
-            print(e)
+        except (requests.exceptions.RequestException, ssl.SSLError):
```

Comment:
why do you remove all other exceptions?

Reply:
The other exceptions are subclasses of `RequestException`, and their exception messages contain enough detail to make special handling for them unnecessary.